### ISSUE TITLE
[pre-sve] ESP32 OTA-R commissioning failure during ReadCommissioningInfo

### ISSUE THREAD

**Issue by s-mcclain** [project-chip/connectedhomeip#19694]
#### Problem
When attempting to commission an OTA-R with the chip-tool, I'm consistently getting commissioning failures due to an Interaction Model Error.

The OTA-R is an esp32 device and I'm using chip-tool on macOS.

The logs are as followed on chip-tool:

```
[1655408434.086047][4838:4840] CHIP:DMG: ReportDataMessage =
[1655408434.086154][4838:4840] CHIP:DMG: {
[1655408434.086263][4838:4840] CHIP:DMG: 	AttributeReportIBs =
[1655408434.086377][4838:4840] CHIP:DMG: 	[
[1655408434.086493][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.086607][4838:4840] CHIP:DMG: 		{
[1655408434.086765][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.087011][4838:4840] CHIP:DMG: 			{
[1655408434.088501][4838:4840] CHIP:DMG: 				DataVersion = 0x8a696b76,
[1655408434.089619][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.090460][4838:4840] CHIP:DMG: 				{
[1655408434.091408][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.094518][4838:4840] CHIP:DMG: 					Cluster = 0x31,
[1655408434.095324][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0003,
[1655408434.096326][4838:4840] CHIP:DMG: 				}
[1655408434.097814][4838:4840] CHIP:DMG:
[1655408434.097943][4838:4840] CHIP:DMG: 					Data = 30,
[1655408434.099537][4838:4840] CHIP:DMG: 			},
[1655408434.100047][4838:4840] CHIP:DMG:
[1655408434.101577][4838:4840] CHIP:DMG: 		},
[1655408434.101716][4838:4840] CHIP:DMG:
[1655408434.101823][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.101929][4838:4840] CHIP:DMG: 		{
[1655408434.102038][4838:4840] CHIP:DMG: 			AttributeStatusIB =
[1655408434.102143][4838:4840] CHIP:DMG: 			{
[1655408434.102252][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.102435][4838:4840] CHIP:DMG: 				{
[1655408434.103432][4838:4840] CHIP:DMG: 					Endpoint = 0xfffe,
[1655408434.103607][4838:4840] CHIP:DMG: 					Cluster = 0x31,
[1655408434.103730][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0003,
[1655408434.103862][4838:4840] CHIP:DMG: 				}
[1655408434.104009][4838:4840] CHIP:DMG:
[1655408434.104148][4838:4840] CHIP:DMG: 				StatusIB =
[1655408434.104282][4838:4840] CHIP:DMG: 				{
[1655408434.104776][4838:4840] CHIP:DMG: 					status = 0x01,
[1655408434.104928][4838:4840] CHIP:DMG: 				},
[1655408434.105080][4838:4840] CHIP:DMG:
[1655408434.105190][4838:4840] CHIP:DMG: 			},
[1655408434.105299][4838:4840] CHIP:DMG:
[1655408434.105402][4838:4840] CHIP:DMG: 		},
[1655408434.105512][4838:4840] CHIP:DMG:
[1655408434.105617][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.105723][4838:4840] CHIP:DMG: 		{
[1655408434.105828][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.105998][4838:4840] CHIP:DMG: 			{
[1655408434.106128][4838:4840] CHIP:DMG: 				DataVersion = 0x94fc7ae6,
[1655408434.106268][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.106396][4838:4840] CHIP:DMG: 				{
[1655408434.106686][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.106927][4838:4840] CHIP:DMG: 					Cluster = 0x28,
[1655408434.107720][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0004,
[1655408434.107818][4838:4840] CHIP:DMG: 				}
[1655408434.107903][4838:4840] CHIP:DMG:
[1655408434.107996][4838:4840] CHIP:DMG: 					Data = 32776,
[1655408434.108083][4838:4840] CHIP:DMG: 			},
[1655408434.108163][4838:4840] CHIP:DMG:
[1655408434.108239][4838:4840] CHIP:DMG: 		},
[1655408434.108321][4838:4840] CHIP:DMG:
[1655408434.108398][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.108476][4838:4840] CHIP:DMG: 		{
[1655408434.108718][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.108845][4838:4840] CHIP:DMG: 			{
[1655408434.108951][4838:4840] CHIP:DMG: 				DataVersion = 0x94fc7ae6,
[1655408434.109018][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.109365][4838:4840] CHIP:DMG: 				{
[1655408434.109441][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.109544][4838:4840] CHIP:DMG: 					Cluster = 0x28,
[1655408434.109616][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0002,
[1655408434.109680][4838:4840] CHIP:DMG: 				}
[1655408434.109750][4838:4840] CHIP:DMG:
[1655408434.109822][4838:4840] CHIP:DMG: 					Data = 65521,
[1655408434.109886][4838:4840] CHIP:DMG: 			},
[1655408434.109948][4838:4840] CHIP:DMG:
[1655408434.110007][4838:4840] CHIP:DMG: 		},
[1655408434.110071][4838:4840] CHIP:DMG:
[1655408434.110154][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.110256][4838:4840] CHIP:DMG: 		{
[1655408434.110321][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.110403][4838:4840] CHIP:DMG: 			{
[1655408434.110718][4838:4840] CHIP:DMG: 				DataVersion = 0x5c9631cb,
[1655408434.111072][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.111175][4838:4840] CHIP:DMG: 				{
[1655408434.111264][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.111351][4838:4840] CHIP:DMG: 					Cluster = 0x30,
[1655408434.111438][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0003,
[1655408434.111523][4838:4840] CHIP:DMG: 				}
[1655408434.111605][4838:4840] CHIP:DMG:
[1655408434.111691][4838:4840] CHIP:DMG: 					Data = 0,
[1655408434.111771][4838:4840] CHIP:DMG: 			},
[1655408434.111850][4838:4840] CHIP:DMG:
[1655408434.111925][4838:4840] CHIP:DMG: 		},
[1655408434.112004][4838:4840] CHIP:DMG:
[1655408434.112085][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.112163][4838:4840] CHIP:DMG: 		{
[1655408434.112242][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.112320][4838:4840] CHIP:DMG: 			{
[1655408434.112403][4838:4840] CHIP:DMG: 				DataVersion = 0x5c9631cb,
[1655408434.112487][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.112567][4838:4840] CHIP:DMG: 				{
[1655408434.112649][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.112735][4838:4840] CHIP:DMG: 					Cluster = 0x30,
[1655408434.112822][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0002,
[1655408434.112905][4838:4840] CHIP:DMG: 				}
[1655408434.112985][4838:4840] CHIP:DMG:
[1655408434.113069][4838:4840] CHIP:DMG: 					Data = 0,
[1655408434.113149][4838:4840] CHIP:DMG: 			},
[1655408434.113227][4838:4840] CHIP:DMG:
[1655408434.113302][4838:4840] CHIP:DMG: 		},
[1655408434.113382][4838:4840] CHIP:DMG:
[1655408434.113459][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.113539][4838:4840] CHIP:DMG: 		{
[1655408434.113618][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.113699][4838:4840] CHIP:DMG: 			{
[1655408434.113781][4838:4840] CHIP:DMG: 				DataVersion = 0x5c9631cb,
[1655408434.113865][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.113954][4838:4840] CHIP:DMG: 				{
[1655408434.114035][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.114121][4838:4840] CHIP:DMG: 					Cluster = 0x30,
[1655408434.114207][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0001,
[1655408434.114291][4838:4840] CHIP:DMG: 				}
[1655408434.114371][4838:4840] CHIP:DMG:
[1655408434.114456][4838:4840] CHIP:DMG: 					Data =
[1655408434.114537][4838:4840] CHIP:DMG: 					{
[1655408434.114623][4838:4840] CHIP:DMG: 						0x0 = 60,
[1655408434.114721][4838:4840] CHIP:DMG: 						0x1 = 900,
[1655408434.114804][4838:4840] CHIP:DMG: 					},
[1655408434.115206][4838:4840] CHIP:DMG: 			},
[1655408434.115300][4838:4840] CHIP:DMG:
[1655408434.115362][4838:4840] CHIP:DMG: 		},
[1655408434.115462][4838:4840] CHIP:DMG:
[1655408434.115527][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.115592][4838:4840] CHIP:DMG: 		{
[1655408434.115654][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.115716][4838:4840] CHIP:DMG: 			{
[1655408434.115782][4838:4840] CHIP:DMG: 				DataVersion = 0x5c9631cb,
[1655408434.115848][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.115912][4838:4840] CHIP:DMG: 				{
[1655408434.115982][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.116051][4838:4840] CHIP:DMG: 					Cluster = 0x30,
[1655408434.116121][4838:4840] CHIP:DMG: 					Attribute = 0x0000_0000,
[1655408434.116244][4838:4840] CHIP:DMG: 				}
[1655408434.116314][4838:4840] CHIP:DMG:
[1655408434.116382][4838:4840] CHIP:DMG: 					Data = 0,
[1655408434.116991][4838:4840] CHIP:DMG: 			},
[1655408434.117288][4838:4840] CHIP:DMG:
[1655408434.117496][4838:4840] CHIP:DMG: 		},
[1655408434.117587][4838:4840] CHIP:DMG:
[1655408434.117666][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.117746][4838:4840] CHIP:DMG: 		{
[1655408434.117811][4838:4840] CHIP:DMG: 			AttributeDataIB =
[1655408434.117887][4838:4840] CHIP:DMG: 			{
[1655408434.117972][4838:4840] CHIP:DMG: 				DataVersion = 0x8a696b76,
[1655408434.118062][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.118269][4838:4840] CHIP:DMG: 				{
[1655408434.118466][4838:4840] CHIP:DMG: 					Endpoint = 0x0,
[1655408434.118542][4838:4840] CHIP:DMG: 					Cluster = 0x31,
[1655408434.118645][4838:4840] CHIP:DMG: 					Attribute = 0x0000_FFFC,
[1655408434.118723][4838:4840] CHIP:DMG: 				}
[1655408434.118807][4838:4840] CHIP:DMG:
[1655408434.119374][4838:4840] CHIP:DMG: 					Data = 1,
[1655408434.119576][4838:4840] CHIP:DMG: 			},
[1655408434.119659][4838:4840] CHIP:DMG:
[1655408434.119760][4838:4840] CHIP:DMG: 		},
[1655408434.119843][4838:4840] CHIP:DMG:
[1655408434.119914][4838:4840] CHIP:DMG: 		AttributeReportIB =
[1655408434.119985][4838:4840] CHIP:DMG: 		{
[1655408434.120055][4838:4840] CHIP:DMG: 			AttributeStatusIB =
[1655408434.120125][4838:4840] CHIP:DMG: 			{
[1655408434.120197][4838:4840] CHIP:DMG: 				AttributePathIB =
[1655408434.120270][4838:4840] CHIP:DMG: 				{
[1655408434.120348][4838:4840] CHIP:DMG: 					Endpoint = 0xfffe,
[1655408434.120448][4838:4840] CHIP:DMG: 					Cluster = 0x31,
[1655408434.120569][4838:4840] CHIP:DMG: 					Attribute = 0x0000_FFFC,
[1655408434.120645][4838:4840] CHIP:DMG: 				}
[1655408434.120735][4838:4840] CHIP:DMG:
[1655408434.120845][4838:4840] CHIP:DMG: 				StatusIB =
[1655408434.120919][4838:4840] CHIP:DMG: 				{
[1655408434.121016][4838:4840] CHIP:DMG: 					status = 0x01,
[1655408434.121118][4838:4840] CHIP:DMG: 				},
[1655408434.121210][4838:4840] CHIP:DMG:
[1655408434.121297][4838:4840] CHIP:DMG: 			},
[1655408434.121387][4838:4840] CHIP:DMG:
[1655408434.121472][4838:4840] CHIP:DMG: 		},
[1655408434.121562][4838:4840] CHIP:DMG:
[1655408434.121647][4838:4840] CHIP:DMG: 	],
[1655408434.121755][4838:4840] CHIP:DMG:
[1655408434.121843][4838:4840] CHIP:DMG: 	SuppressResponse = true,
[1655408434.122194][4838:4840] CHIP:DMG: 	InteractionModelRevision = 1
[1655408434.122310][4838:4840] CHIP:DMG: }
[1655408434.122932][4838:4840] CHIP:CTL: Error parsing commissioning information
[1655408434.123103][4838:4840] CHIP:CTL: Error on commissioning step 'ReadCommissioningInfo': 'CHIP Error 0x000000CA: Interaction Model Error'
[1655408434.123206][4838:4840] CHIP:CTL: Failed to perform commissioning step 2
[1655408434.123317][4838:4840] CHIP:CTL: Going from commissioning step 'ReadCommissioningInfo' with lastErr = 'CHIP Error 0x000000CA: Interaction Model Error' -> 'Cleanup'
```

On the esp32:

```
I (369185) chip[DL]: Confirm received for CHIPoBLE TX characteristic indication (con 1) status= 0
I (369285) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 1 12
I (369285) chip[EM]: Received message of type 0x22 with protocolId (0, 0) and MessageCounter:3983196915 on exchange 17247r
I (370315) chip[IN]: Prepared unauthenticated message 0x3fcb0df8 to 0x0000000000000000 (0)  of type 0x23 and protocolId (0, 0) on exchange 17247r with MessageCounter:84079076.
I (370315) chip[IN]: Sending unauthenticated msg 0x3fcb0df8 with MessageCounter:84079076 to 0x0000000000000000 at monotonic time: 000000000005A693 msec
I (370335) NimBLE: GATT procedure initiated: notify;
I (370335) NimBLE: att_handle=14

I (370345) chip[DL]: Confirm received for CHIPoBLE TX characteristic indication (con 1) status= 0
E (370355) chip[DL]: Long dispatch time: 1065 ms, for event type 7
I (370455) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 1 12
I (370455) chip[EM]: Received message of type 0x24 with protocolId (0, 0) and MessageCounter:3983196916 on exchange 17247r
I (370465) chip[IN]: Prepared unauthenticated message 0x3fcb0e88 to 0x0000000000000000 (0)  of type 0x40 and protocolId (0, 0) on exchange 17247r with MessageCounter:84079077.
I (370475) chip[IN]: Sending unauthenticated msg 0x3fcb0e88 with MessageCounter:84079077 to 0x0000000000000000 at monotonic time: 000000000005A734 msec
I (370495) NimBLE: GATT procedure initiated: notify;
I (370495) NimBLE: att_handle=14

I (370505) chip[DL]: Confirm received for CHIPoBLE TX characteristic indication (con 1) status= 0
I (370515) chip[SVR]: Commissioning completed session establishment step
I (370525) chip[DIS]: Updating services using commissioning mode 0
I (370525) chip[DIS]: CHIP minimal mDNS started advertising.
I (370535) chip[DL]: NVS set: chip-config/fail-safe-armed = true
I (370545) chip[SVR]: Device completed Rendezvous process
I (370535) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 1 12
I (370555) chip[EM]: Received message of type 0x2 with protocolId (0, 1) and MessageCounter:6788078 on exchange 17248r
I (370575) chip[IN]: Prepared secure message 0x3fcb10b8 to 0xFFFFFFFB00000000 (0)  of type 0x5 and protocolId (0, 1) on exchange 17248r with MessageCounter:51992627.
I (370585) chip[IN]: Sending encrypted msg 0x3fcb10b8 with MessageCounter:51992627 to 0xFFFFFFFB00000000 (0) at monotonic time: 000000000005A79A msec
I (370595) NimBLE: GATT procedure initiated: notify;
I (370605) NimBLE: att_handle=14

I (370605) chip[DL]: Confirm received for CHIPoBLE TX characteristic indication (con 1) status= 0
I (370615) NimBLE: GATT procedure initiated: notify;
I (370625) NimBLE: att_handle=14

I (370625) chip[DL]: Confirm received for CHIPoBLE TX characteristic indication (con 1) status= 0
I (370685) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 1 12
I (370695) chip[EM]: Received message of type 0x8 with protocolId (0, 1) and MessageCounter:6788079 on exchange 17249r
I (370695) chip[FS]: GeneralCommissioning: Received ArmFailSafe (0s)
I (370705) chip[FS]: Fail-safe timer expired
```

**Comment by bzbarsky-apple** [comment]
I believe this is partial fallout for the fix  #19428 @carol-apple for the "linux" version of the app that added this secondary endpoint.  The ESP32 app should be disabling the secondary network commissioning endpoint that it's not actually using.  `Esp32AppServer::Init` sets up a network commissioning driver on endpoint 0, but does not disable the secondary endpoint.

`examples/all-clusters-app/esp32/main/main.cpp` does 
```
    emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, false);
```
in `InitServer` right after `Esp32AppServer::Init`.  It looks like `examples/ota-provider-app/esp32/main/main.cpp` should do the same.

**Comment by shubhamdp** [comment]
Thanks @bzbarsky-apple, I have put up a PR to fix this.
